daffodil-lib/src/main/scala/org/apache/daffodil/lib/xml/XMLUtils.scala

```diff
@@ -1459,34 +1459,27 @@ Differences were (path, expected, actual):
           )
       }
 
-    val uriIsJustPathComponent =
-      uri.getScheme == null &&
-        uri.getAuthority == null &&
-        uri.getQuery == null &&
-        uri.getFragment == null &&
-        uri.getPath != null
-
     val optResolved: Option[(URISchemaSource, Boolean)] =
-      if (uri.isAbsolute) {
-        // an absolute URI is one with a scheme. In this case, we expect to be able to resolve
-        // the URI and do not try anything else (e.g. filesystem, classpath). Since this function
+      if (uri.isAbsolute && uri.getScheme.contains("jar")) {
```

Comment:
This should be `uri.getScheme == "jar"`. Avoids ever accidentally matching some other scheme like "foojar" if that ever happens.